src/observable/Collection.js

```diff
@@ -49,9 +49,15 @@ define([
             this.onChanged.dispatch([this, EVENTS.ITEM_ADDED, item]);
         },
         remove: function(item) {
-            this._remove(item);
-            this._collection.splice(this._collection.indexOf(item), 1);
```

Comment:
If item was not found, this would result in splice(-1,1) which would just remove the last item in the array. :facepunch: